interfaces/builtin/home.go

```diff
@@ -74,6 +74,11 @@ owner @{HOME}/snap/ r,
 # files; only allow writes to files, not the mount point).
 owner /run/user/[0-9]*/gvfs/{,**} r,
 owner /run/user/[0-9]*/gvfs/*/**  w,
+
+# On some systems HOME/bin is in the default path in ahead of system
+# directories. To prevent snaps from writing arbitrary executables waiting to
+# be launched by the user, deny writes to the directory altogether.
+deny @{HOME}/bin/{,**} w,
```

Comment:
Please add 'l' otherwise they can create hardlinks in here to a file they control.